polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td

```diff
@@ -25,4 +25,10 @@ class Polygeist_Type<string name, string typeMnemonic, list<Trait> traits = []>
   let mnemonic = typeMnemonic;
 }
 
+def Polygeist_StructType :
```

Comment:
This type needs to implement the `MemRefElementTypeInterface` to indicate it can be legally used as the lement type of a `MemRefType`.

https://mlir.llvm.org/docs/Dialects/Builtin/#memreftype

Reply:
Already does on line 22.